staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -216,7 +215,7 @@ func (s *Serializer) Encode(obj runtime.Object, w io.Writer) error {
 		_, err = w.Write(data)
 		return err
 	}
-	encoder := json.NewEncoder(w)
+	encoder := jsoniter.ConfigCompatibleWithStandardLibrary.NewEncoder(w)
```

Comment:
Everything looks OK except this.  Doesn't this have a negative impact of performance?  It did when I measured it.

Reply:
I offered a bench commit, on my machine it's
```
pkg: k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json
BenchmarkJsonEncoding-4       	 3000000	       569 ns/op
BenchmarkJsoniterEncoding-4   	 5000000	       316 ns/op
```
How did you measure it?

Reply:
The benchmark doesn't use data that is similar to actual objects; can you get e.g. a big pod spec or something and run again?

Reply:
@lavalamp  I pushed a commit, the result was not better, but just 20% improved.
```
pkg: k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json
BenchmarkJsonEncoding-4       	  100000	     13080 ns/op
BenchmarkJsoniterEncoding-4   	  200000	     11158 ns/op
PASS
```